### DISCUSSION TITLE
How to execute a block of commands in their own scope?

### DISCUSSION THREAD

**Discussion by HitomiTenshi** [nushell/nushell#6323] [Q&A] [answered]
In the documentation for [Getting Started -> Thinking in Nu -> Nushell's environment is scoped](https://www.nushell.sh/book/thinking_in_nu.html#nushell-s-environment-is-scoped) there is a scoped each block example. Each iterated item starts with a fresh scope of the original working directory.

```shell
> ls | each { |it|
    cd $it.name
    make
}
```

How do I achieve the same thing without an each iteration?

Example, execute 2 blocks in their own scope:

```shell
cd my-folder
ls
```

After that execute another block with its own scope:

```shell
cd my-other-folder
ls
```

**Comment by panicbit** [comment, marked as answer]
```
do { cd my-other-folder; ls }
```
![grafik](https://user-images.githubusercontent.com/628445/184550803-371153d3-7f84-42ea-9395-c724a8a82507.png)

**Reply by HitomiTenshi** [reply]
Sweet, exactly what I was looking for. Thanks!
